app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
Ok, I'm just concerned about the growing complexity of our controllers. We should have a common strategy while adding JSON API endpoints. For instance taking advantage of this cases to simplify them. The risk is that we'll never do that in the future and understanding this classes will be difficult.

But again, if I'm the only one concerned... go ahead 👍